drivers/sensors/ltr308.c

```diff
@@ -0,0 +1,1000 @@
+            (FAR struct ltr308_threshold_s *)arg;
+          ret = ltr308_get_threshold(priv, &data->upper_limit,
+                                     &data->lower_limit);
+        }
+        break;
+
+      case SNIOC_GET_LUX:
+        {
+          FAR struct ltr308_lux_s *data = (FAR struct ltr308_lux_s *)arg;
+          ret = ltr308_get_lux(priv, data->gain, data->integration_time,
+                               data->ch, &data->lux);
+        }
+        break;
+
+      default:
+        {
+          snerr("Unrecognized cmd: %d\n", cmd);
+          ret = -ENOTTY;
+        }
+        break;
+    }
+
+  nxsem_post(&priv->devsem);
+  return ret;
+}
+
+/****************************************************************************
+ * Name: ltr308_unlink
+ *
+ * Description:
+ *   Standard character driver unlink method.
+ *
+ ****************************************************************************/
+
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+static int ltr308_unlink(FAR struct inode *inode)
+{
+  FAR struct ltr308_dev_s *priv;
+  int ret;
+
+  DEBUGASSERT(inode != NULL && inode->i_private != NULL);
+  priv = (FAR struct ltr308_dev_s *)inode->i_private;
+
+  /* Get exclusive access */
+
+  ret = nxsem_wait_uninterruptible(&priv->devsem);
+  if (ret < 0)
+    {
+      snerr("nxsem_wait_uninterruptible (err = %d)\n", ret);
+      return ret;
+    }
+
+  /* Check if there any open references to the driver */
+
+  if (priv->crefs <= 0)
+    {
+      nxsem_destroy(&priv->devsem);
+      kmm_free(priv);
+      return OK;
+    }
+
+  /*
+   * Mark the driver as unlinked and free the resources when the last client
+   * closes their reference to the driver.
+   */
+
+  priv->unlinked = true;
+  nxsem_post(&priv->devsem);
+  return OK;
+}
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: ltr308_register
+ *
+ * Description:
+ *   Register the LTR308 character device as 'devpath'
+ *
+ * Input Parameters:
+ *   devpath - The full path to the driver to register. E.g., "/dev/light0"
+ *   i2c     - An instance of the I2C interface to use to communicate with
+ *             LTR308
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int ltr308_register(const char *devpath, FAR struct i2c_master_s *i2c)
+{
+  FAR struct ltr308_dev_s *priv;
+  int ret = 0;
+
+  DEBUGASSERT(i2c != NULL);
+
+  /* Initialize the LTR308 device structure */
+
+  priv = (FAR struct ltr308_dev_s *)kmm_zalloc(sizeof(struct ltr308_dev_s));
+  if (!priv)
+    {
+      snerr("ERROR: Failed to allocate instance\n");
+      return -ENOMEM;
+    }
+
+  priv->i2c = i2c;
+  priv->addr = LTR308_ADDR;
```

Comment:
@pkarashchenko should I remove the "addr" field from the private driver structure and use only the define, since the sensor's address is fixed?

Reply:
Yeah. Using define directly and reducing size of priv structure is welcomed